### ISSUE TITLE
Bug founds in testing to analyze

### ISSUE THREAD

**Issue by xurxodev** [EyeSeeTea/metadata-synchronization#337]
- [x] When to realize logout in dhis or session is finished, metadata sync app may do crash showing javascript error,  example use case: to do login in dhis and in app navigate to aggregated manual sync then logout from dhis and in the app back to landing and click in either link => Move to #371

- [x] Infinite progress without message to user to sync events in Org unit step if user has no assigned org unit
- [x] Infinite progress without message to user to sync events in events step if user has no permissions
- [x] Edit action is disabled as the primary action only when the user has not METADATA_SYNC_CONFIGURATOR role but if the user can not edit the sync rule by share settings the edit action is the primary action and edit action should not be the primary action in this case. => Move to #356

**Comment by xurxodev** [comment]
@adrianq I have analyzed the behavior with finished dhis2 session in a production version of the app.
When a dhis session is finished, the javascript errors appears in the console and the UI is freezes or with infinity progress bars to interact with the app.

I have tested dhis2 apps as event capture or maintenance on this same situation and inside the apps the behavior is similar, javascript errors appears in the console and the UI is freezes or with infinity progress bars to interact with the app. But when without a session the user try enter in a new app or you refresh the url, the ui is redirect to login.

I think that one possible solution could be that we can use the new authorization component to implement without session behavior in routes, for example redirecting to login if user is not logged. We should add the authorization component as parent of all routes in the root component and to use a function to know is user is logged.
